pkg/controller/controller.go

```diff
@@ -98,17 +112,38 @@ func NewResizeController(
 		DeleteFunc: ctrl.deletePVC,
 	}, resyncPeriod)
 
+	podInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
+		AddFunc:    ctrl.addPod,
+		DeleteFunc: ctrl.deletePod,
+	}, resyncPeriod)
+
 	return ctrl
 }
 
 func (ctrl *resizeController) addPVC(obj interface{}) {
-	objKey, err := getPVCKey(obj)
+	objKey, err := getObjectKey(obj)
 	if err != nil {
 		return
 	}
 	ctrl.claimQueue.Add(objKey)
 }
 
+func (ctrl *resizeController) addPod(obj interface{}) {
+	pod := parsePod(obj)
+	if pod != nil {
```

Comment:
Should this be "if pod == nil"?  Otherwise pod is nil on line 136.

Reply:
good catch. thank you.

Reply:
this is fixed. Also added tests that cover this behaviour.